src/fmeval/transforms/common.py

```diff
@@ -23,9 +29,12 @@ def __init__(self, input_keys: List[str], output_keys: List[str], prompt_templat
         :param output_keys: The keys corresponding to the prompts that get added by this Transform.
         :param prompt_template: The template used to construct the prompt.
             Example: "Summarize the following text: $model_input".
+        :param placeholder_keys_dict: The placeholders and the corresponding input keys dict.
+            Example: {"question": "model_input"} with prompt_template "Question: $question"
```

Comment:
I think that fleshing out a full example underneath the parameter docstrings will make the usage much clearer, as this is admittedly quite confusing at a glance. For example:

```
Example usage:

Inputs:
prompt_template = "Summarize $x and $y"
output_keys = ["my_prompt"] # note that when using placeholder_keys_dict, having more than one output key doesn't make much sense, as all of the composed prompts will be identical.
placeholder_keys_dict = {"x": "statement_1", "y": "statement_2"}
record = {"statement_1": "some long text", "statement_2": "some other long text"}

Output record (only new keys and values are shown):
{"my_prompt": "Summarize some long text and some other long text"}
```

I would make sure to highlight in the docstrings for the parameters that `input_keys` doesn't get used when `placeholder_keys_dict` gets used, and additionally that it only makes sense to have multiple `output_keys` when not using `placeholder_keys_dict`.